src/faction_camp.cpp

```diff
@@ -85,7 +88,7 @@ static std::array<std::pair<const std::string, point>, COMPANION_SORT_POINTS> so
     }
 };
 
-static bool update_time_left( std::string &entry, const std::shared_ptr<npc> comp )
+static bool update_time_left( std::string &entry, const npc_ptr comp )
```

Comment:
A reference to `npc` would be much better than a pointer. Currently the pointer is derferenced without any check for it being null. It is clearly expected to never be null. Using a reference as parameter type makes this clear to the caller.

Same with the other functions below.

Reply:
Fixed.